hazelcast/src/test/java/com/hazelcast/cache/MapConfigTest.java

```diff
@@ -0,0 +1,76 @@
+package com.hazelcast.cache;
+
+import com.google.common.collect.Lists;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.EntryListenerConfig;
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.MaxSizeConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class MapConfigTest {
+    private HazelcastInstance hazelcast;
+
+    @Before
+    public void setUp() {
+        Config hazelcastConfig = new Config();
+        hazelcastConfig.addMapConfig(new MapConfig("my.custom.map.*")
+                .setMaxSizeConfig(new MaxSizeConfig(5000, MaxSizeConfig.MaxSizePolicy.PER_NODE))
+                .setTimeToLiveSeconds(3600)
+                .setBackupCount(1)
+                .setMaxIdleSeconds(3600)
+                .setEvictionPolicy(EvictionPolicy.LRU)
+                .setAsyncBackupCount(0)
+                .setInMemoryFormat(InMemoryFormat.BINARY)
+                .setReadBackupData(false)
+                .setNearCacheConfig(null)
+                .setMapStoreConfig(null)
+                .setWanReplicationRef(null)
+                .setEntryListenerConfigs(Lists.<EntryListenerConfig>newArrayList())
+                .setMapIndexConfigs(Lists.<MapIndexConfig>newArrayList())
+                .setMergePolicy("com.hazelcast.map.merge.PutIfAbsentMapMergePolicy"));
+        hazelcast = Hazelcast.newHazelcastInstance(hazelcastConfig);
+    }
+
+    @After
+    public void tearDown() {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void testMapConfigShouldPrependWilcardConfig() {
```

Comment:
This test is passing in ```3.8.6```, but failing in ```3.9.1```. The problem is that static wildcard settings may prepend actual map settings, created after hazelcast instance is started. That looks wrong, and is fixed in this pull request